src/main/java/frc/trigon/robot/poseestimation/poseestimator/PoseEstimatorConstants.java

```diff
@@ -1,18 +1,50 @@
 package frc.trigon.robot.poseestimation.poseestimator;
 
-import edu.wpi.first.math.VecBuilder;
-import edu.wpi.first.math.Vector;
-import edu.wpi.first.math.numbers.N3;
+import edu.wpi.first.math.geometry.Transform2d;
 
 public class PoseEstimatorConstants {
     public static final double ODOMETRY_FREQUENCY_HERTZ = 250;
 
+    static final double POSE_BUFFER_SIZE_SECONDS = 2;
+
+    static final StandardDeviations ODOMETRY_STANDARD_DEVIATIONS = new StandardDeviations(0.003, 0.0002);
+
+
     /**
-     * The vector represents how ambiguous each value of the odometry is.
-     * The first value represents how ambiguous is the x,
-     * the second one for the y, and the third one is for the theta (rotation).
-     * Increase these numbers to trust the estimate less.
+     * A record that stores how ambiguous the estimated pose of the robot is.
+     * The greater the number, the less trustworthy the pose is.
+     *
+     * @param translation the ambiguity of the translation aspect of the pose estimation
+     * @param theta       the ambiguity of the rotation aspect of the pose estimation
      */
-    static final Vector<N3> ODOMETRY_AMBIGUITY = VecBuilder.fill(0.003, 0.003, 0.0002);
+    public record StandardDeviations(double translation, double theta) {
+        PoseEstimatorConstants.StandardDeviations combineOdometryAndVisionStandardDeviations() {
+            final PoseEstimatorConstants.StandardDeviations odometryStandardDeviations = PoseEstimatorConstants.ODOMETRY_STANDARD_DEVIATIONS;
+
+            return new PoseEstimatorConstants.StandardDeviations(
+                    combineOdometryAndVisionStandardDeviation(odometryStandardDeviations.translation(), this.translation()),
+                    combineOdometryAndVisionStandardDeviation(odometryStandardDeviations.theta(), this.theta())
+            );
+        }
+
+        Transform2d scaleTransformFromStandardDeviations(Transform2d transform) {
+            return new Transform2d(
+                    transform.getX() * this.translation(),
+                    transform.getY() * this.translation(),
+                    transform.getRotation().times(this.theta())
+            );
+        }
+
+        /**
+         * Combines a standard deviation value of the odometry and of the vision result to get a new standard deviation.
+         *
+         * @param odometryStandardDeviation a standard deviation value of the estimated odometry pose
+         * @param visionStandardDeviation   a standard deviation value of the estimated vision pose
+         * @return the combined standard deviation
+         */
+        private double combineOdometryAndVisionStandardDeviation(double odometryStandardDeviation, double visionStandardDeviation) {
+            return odometryStandardDeviation / (odometryStandardDeviation + Math.sqrt(odometryStandardDeviation * visionStandardDeviation));
+        }
```

Comment:
1. it is inefficient to create a new odom stddevs every time this class is created (which also suggests some recursion)
2.  assume this is how you do it for each stddev for now, and remove the odom part for more abstraction